sdk/storage/Azure.Storage.Common/tests/Shared/TestHelper.cs

```diff
@@ -149,5 +150,17 @@ public static void AssertCacheableProperty<T>(T expected, Func<T> property)
             Assert.AreEqual(expected, actual); // first call calculates and caches value
             Assert.AreSame(actual, property()); // subsequent calls use cached value
         }
+
+        public static void AssertInconclusiveRecordingFriendly(RecordedTestMode mode, string message = default)
```

Comment:
What is this for?

Reply:
We want to assert inconclusive for some tests. A few of the inherited tests don't apply to specific clients and so those clients assert inconclusive when asked to do something they can't. However, Inconclusive tests aren't recorded. This makes playback an issue. With this, we assert success only when in record mode so that a recording file is generated, allowing tests to then actually assert inconclusive in playback instead of failing due to lack of recording.